src/architecture/messaging/_UnitTest/test_unsubscribe.py

```diff
@@ -0,0 +1,105 @@
+#
+#  Copyright (c) 2025, Autonomous Vehicle Systems Lab, University of Colorado at Boulder
+#
+#  Permission to use, copy, modify, and/or distribute this software for any
+#  purpose with or without fee is hereby granted, provided that the above
+#  copyright notice and this permission notice appear in all copies.
+#
+#  THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+#  WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+#  MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+#  ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+#  WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+#  ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+#  OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+#
+import time
+from typing import Union
+
+from Basilisk.moduleTemplates import cModuleTemplate
+from Basilisk.moduleTemplates import cppModuleTemplate
+from Basilisk.architecture import messaging
+from Basilisk.architecture import bskLogging
+
+def assertConnectionWithValues(
+    inMsg: Union[messaging.CModuleTemplateMsg_C, messaging.CModuleTemplateMsgReader],
+    outMsg: Union[messaging.CModuleTemplateMsg_C, messaging.CModuleTemplateMsg]
+):
+    val = time.time() # unique value
+
+    payload = messaging.CModuleTemplateMsgPayload()
+    payload.dataVector = [val, 0, 0]
+    outMsg.write(payload)
+
+    if isinstance(inMsg, messaging.CModuleTemplateMsg_C):
+        readPayload: messaging.CModuleTemplateMsgPayload = inMsg.read()
+    else:
+        readPayload: messaging.CModuleTemplateMsgPayload = inMsg()
+
+    assert readPayload.dataVector[0] == val
+
+def test_unsubscribe():
+    """
+    testing the unsubscribe functions of C and C++ messages
+    """
+    # create modules
+    cMod = cModuleTemplate.cModuleTemplate()
+    cppMod = cppModuleTemplate.CppModuleTemplate()
+
+    # check that unsubscribe is a noop for unlinked OnMsgs
+    cMod.dataInMsg.unsubscribe()
+    cppMod.dataInMsg.unsubscribe()
+
+    # check that unsubscribe is noop for (C) OutMsg
+    cMod.dataOutMsg.unsubscribe()
```

Comment:
What happens if multiple modules are redirected to write to a single external stand-alone message?  If we call unsubscribe to the OutMsg, is just the out msg of the BSK module disconnected, or are all re-directions to the stand-alone msg disconnected?